src/main/java/seedu/address/storage/DrugSearchUtility.java

```diff
@@ -0,0 +1,126 @@
+package seedu.address.storage;
+
+import com.opencsv.CSVReader;
+import com.opencsv.CSVReaderBuilder;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+
+
+/**
+ * For searching through "src/main/resources/storage/datasetForDrugs.csv"
+ * for drugs whose names contain a certain keyword, then
+ * returning pharmacological information about those drugs.
+ */
+
+public class DrugSearchUtility {
+    private static final String DATASET_PATH = "./src/main/resources/storage/datasetForDrugs.csv"; //filepath to drug dataset
+
+    private static ArrayList<String[]> resultsCache = new ArrayList<>(); //cached results of most recent keyword search
+
+    private static String[] tooGeneric = {"capsule","cream","emulsi","gel","hydrochloride",
+            "injection","lotion","ointment","paste","patch","pill","powder",
+            "solution","supposit","syrup","tablet"};  //keywords too generic to search for, as they will match tens of drugs
+
+    /*
+     * For writing to error log.
+     */
+
+    private static FileWriter f = setF();
+    private static PrintWriter errorLog = new PrintWriter(new BufferedWriter(f));
+
+    /**
+     * Find in "dataSetForDrugs.csv" all drugs whose names
+     * contain the given keyword.
+     *
+     * @return a formatted String containing the list (and
+     * pharmacological data) of all matching drugs.
+     */
+    public static String find(String keyword) {
+        String results = "";
+        resultsCache.clear(); //clearing cached results from previous search
+
+        for (String i: tooGeneric) {
+            if(i.toLowerCase().contains(keyword.toLowerCase())) {
+                return "Your entered keyword " + keyword + " is too generic, and will lead to too many results. Try a longer keyword," +
+                        " or a more specific one.";
+            }
+        }
+
+        try {
+            Reader reader = Files.newBufferedReader(Paths.get(DATASET_PATH));
+            CSVReader csvReader = new CSVReaderBuilder(reader).withSkipLines(1).build();
+
+           String nextRecord[];
+           while ((nextRecord = csvReader.readNext()) != null) {
+               if(nextRecord[1].toLowerCase().contains(keyword.toLowerCase())) {
+                   resultsCache.add(nextRecord);
+               }
+           }
+
+           for(int i = 0; i < resultsCache.size(); i++) {
+               String[] currentRecord = resultsCache.get(i);
```

Comment:
Use a StringBuilder here for consistency with other parts of the program which perform string manipulation. In this case, we have multiple fields, so we probably want to encapsulate the result under a DrugRecord class with the fields as an enumerated variable, mapped to their values using a HashMap data structure.